shared/lib_battery_dispatch_pvsmoothing_fom.cpp

```diff
@@ -0,0 +1,440 @@
+
+void dispatch_pvsmoothing_front_of_meter_t::init_with_pointer(const dispatch_pvsmoothing_front_of_meter_t* tmp)
+{
+	_forecast_hours = tmp->_forecast_hours;
+	_inverter_paco = tmp->_inverter_paco;
+
+	m_etaPVCharge = tmp->m_etaPVCharge;
+	m_etaGridCharge = tmp->m_etaGridCharge;
+	m_etaDischarge = tmp->m_etaDischarge;
+
+    m_batt_dispatch_pvs_nameplate_ac = tmp->m_batt_dispatch_pvs_nameplate_ac;
+    m_batt_dispatch_pvs_ac_lb = tmp->m_batt_dispatch_pvs_ac_lb;
+    m_batt_dispatch_pvs_ac_lb_enable = tmp->m_batt_dispatch_pvs_ac_lb_enable;
+    m_batt_dispatch_pvs_ac_ub = tmp->m_batt_dispatch_pvs_ac_ub;
+    m_batt_dispatch_pvs_ac_ub_enable = tmp->m_batt_dispatch_pvs_ac_ub_enable;
+    m_batt_dispatch_pvs_curtail_as_control = tmp->m_batt_dispatch_pvs_curtail_as_control;
+    m_batt_dispatch_pvs_curtail_if_violation = tmp->m_batt_dispatch_pvs_curtail_if_violation;
+    m_batt_dispatch_pvs_forecast_shift_periods = tmp->m_batt_dispatch_pvs_forecast_shift_periods;
+    m_batt_dispatch_pvs_kf = tmp->m_batt_dispatch_pvs_kf;
+    m_batt_dispatch_pvs_ki = tmp->m_batt_dispatch_pvs_ki;
+    m_batt_dispatch_pvs_kp = tmp->m_batt_dispatch_pvs_kp;
+    m_batt_dispatch_pvs_max_ramp = tmp->m_batt_dispatch_pvs_max_ramp;
+    m_batt_dispatch_pvs_short_forecast_enable = tmp->m_batt_dispatch_pvs_short_forecast_enable;
+    m_batt_dispatch_pvs_soc_rest = tmp->m_batt_dispatch_pvs_soc_rest;
+    m_batt_dispatch_pvs_timestep_multiplier = tmp->m_batt_dispatch_pvs_timestep_multiplier;
+
+}
+
+
+// deep copy from dispatch to this
+dispatch_pvsmoothing_front_of_meter_t::dispatch_pvsmoothing_front_of_meter_t(const dispatch_t & dispatch) :
+dispatch_automatic_t(dispatch)
+{
+	const dispatch_pvsmoothing_front_of_meter_t * tmp = dynamic_cast<const dispatch_pvsmoothing_front_of_meter_t *>(&dispatch);
+	init_with_pointer(tmp);
+}
+
+// shallow copy from dispatch to this
+void dispatch_pvsmoothing_front_of_meter_t::copy(const dispatch_t * dispatch)
+{
+	dispatch_automatic_t::copy(dispatch);
+	const dispatch_pvsmoothing_front_of_meter_t * tmp = dynamic_cast<const dispatch_pvsmoothing_front_of_meter_t *>(dispatch);
+	init_with_pointer(tmp);
+}
+
+void dispatch_pvsmoothing_front_of_meter_t::dispatch(size_t year,
+	size_t hour_of_year,
+	size_t step)
+{
+    curr_year = year;
+	size_t step_per_hour = (size_t)(1 / _dt_hour);
+	size_t lifetimeIndex = util::lifetimeIndex(year, hour_of_year, step, step_per_hour);
+
+	update_dispatch(year, hour_of_year, step, lifetimeIndex);
+	dispatch_automatic_t::dispatch(year, hour_of_year, step);
+}
+
+void dispatch_pvsmoothing_front_of_meter_t::update_dispatch(size_t year, size_t hour_of_year, size_t , size_t lifetimeIndex)
+{
+	// Initialize
+	m_batteryPower->powerBatteryDC = 0;
+	m_batteryPower->powerBatteryAC = 0;
+	m_batteryPower->powerBatteryTarget = 0;
+
+    size_t ndx_sampled = lifetimeIndex / m_batt_dispatch_pvs_timestep_multiplier;
+
+    bool linear_interpolation = false; // updated per 7/14/2021 meeting
```

Comment:
Under what conditions might this be true in the future? If it's unlikely to be used, I'd prefer removing the unused code to leaving it, as I think this implementation (hardcoded to false) is likely to cause confusion.

Reply:
Left in for next meeting review - I prefer to have it removed but they seem to go back and fortth with the step curves in the time series.

Reply:
removed with last commit